### ISSUE TITLE
"pristine" calculation only does a shallow comparison

### ISSUE THREAD

**Issue by begincalendar** [erikras/redux-form#2647]
### Are you submitting a **bug report** or a **feature request**?
bug report

### What is the current behaviour?
[This line](https://github.com/erikras/redux-form/blob/master/src/ConnectedField.js#L216) only does a "shallow" comparison between `value` and `initial` and yet the docs state that values can be any value.

If the values are object-based (e.g. Object, Array, etc.), then the calculation produces incorrect behavior, because a pointer equality is performed and not a deep equality operation.

### What is the expected behavior?
Perform deep equality operation.

### What's your environment?
N/A

### Other informations
N/A

**Comment by erikras** [comment]
Fix published in [`v6.6.0`](https://github.com/erikras/redux-form/releases/tag/v6.6.0).

**Comment by RomeroDiver** [comment]
Hi, I am having a similar bug when using pristine from props - if I bind my initialValues to an object, the pristine flag always returns true. 
E.g.
```
const Test = ({ pristine })=> { ... }
const mapStateToProps = (state) => ({
  initialValues: {
    testProps: { foo: 'bar' } 
  }
})
